src/main/java/gregtech/common/metatileentities/electric/MetaTileEntityPump.java

```diff
@@ -254,14 +283,78 @@ private void tryPumpFirstBlock() {
             }
             FluidStack drainStack = fluidHandler.drain(Integer.MAX_VALUE, false);
             if (drainStack != null && exportFluids.fill(drainStack, false) == drainStack.amount) {
-                exportFluids.fill(drainStack, true);
-                fluidHandler.drain(drainStack.amount, true);
-                this.fluidSourceBlocks.remove(fluidBlockPos);
-                energyContainer.changeEnergy(-GTValues.V[getTier()] * 2);
+                if (locked) {
```

Comment:
```suggestion
                if (locked && !drainStack.isFluidEqual(lockedFluid)) {
```
Prevents unnecessary calls to `FluidStack#copy`

Reply:
checking `lockedFluid == null` also prevents unneeded calls to `copy()` and is what the quantum tank does